micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/web/resttemplate/fluent/common/response/executor/RestExecutor.groovy

```diff
@@ -36,10 +39,17 @@ final class RestExecutor<T> {
         try {
             return exchangeInternal(params, httpMethod, responseType).get()
         } catch (ExecutionException e) {
-            throw e.cause
+            propagate(e.cause)
         }
     }
 
+    private void propagate(Throwable throwable) {
+        if (throwable instanceof TimeoutException) {
```

Comment:
Can we have any other checked exceptions here? If that is the case then we can just always throw our custom runtime exception without checking if it's TimeoutException?